### PR TITLE
[19277] Ensure the effective rect is fetched correctly when screen is locked

### PR DESCRIPTION
**Pull request by livecodepanos**

### CODE REVIEW

File: engine/src/desktop-stack.cpp [hunk 1]
```diff
@@ -225,9 +225,18 @@ void MCStack::realize(void)
...
-	MCRectangle t_rect;
-	MCPlatformGetWindowFrameRect(window, t_rect);
-	return t_rect;
+    MCRectangle t_frame_rect;
+	MCPlatformGetWindowFrameRect(window, t_frame_rect);
+    if (MClockscreen != 0)
+    {
+        MCRectangle t_content_rect, t_diff_rect;
+        MCscreen->platform_getwindowgeometry(window, t_content_rect);
+        t_diff_rect = MCU_subtract_rect(t_content_rect, t_frame_rect);
```
Comment (runrevmark) [inline engine/src/desktop-stack.cpp]: I don't think MCU_subtract_rect is correc there - t_content_rect here is the window's current rect, and t_frame_rect is the window's current effective rect. The frame rect will contain the content rect, which gives you four margins - one for top, left, right and bottom. You then want to apply these margins to the rect of the stack - e.g. left of effective rect = rect.left + (t_content_rect.x - t_frame_rect.x).

**Comment by montegoulding** [comment]
@livecodepanos this looks testable when running in ui mode to me so I'll add the needs tests label

**Comment by montegoulding** [comment]
@livecode-vulcan review ok e90667f

**Comment by livecode-vulcan** [comment]
💙 review by @montegoulding ok e90667f81d4c726a4459e86cc8b0b954d21d800d

**Comment by livecode-vulcan** [comment]
😞 test failure e90667f81d4c726a4459e86cc8b0b954d21d800d

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/223)
* try-community-armv7-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-sdk26_ndk16r15/builds/215)
* try-community-arm64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-sdk26_ndk16r15/builds/215)
* try-community-x86-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-sdk26_ndk16r15/builds/215)
* try-community-x86_64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-sdk26_ndk16r15/builds/215)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1737)
* try-community-universal-ios-iphoneos12.0: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos12.0/builds/119)
* try-community-universal-ios-iphonesimulator12.0: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator12.0/builds/119)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1768)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1760)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1757)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1793)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1734)

**Comment by montegoulding** [comment]
@livecode-vulcan review ok 0011c72

**Comment by livecode-vulcan** [comment]
💙 review by @montegoulding ok 0011c7249fdf2392dc89b25364c00d72cc459340

**Comment by livecode-vulcan** [comment]
😎 test success 0011c7249fdf2392dc89b25364c00d72cc459340

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/224)
* try-community-armv7-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-sdk26_ndk16r15/builds/216)
* try-community-arm64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-sdk26_ndk16r15/builds/216)
* try-community-x86-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-sdk26_ndk16r15/builds/216)
* try-community-x86_64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-sdk26_ndk16r15/builds/216)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1738)
* try-community-universal-ios-iphoneos12.0: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos12.0/builds/120)
* try-community-universal-ios-iphonesimulator12.0: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator12.0/builds/120)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1769)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1761)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1758)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1794)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1735)

**Pull request merged**